src/Libraries/Microsoft.Extensions.AI.Abstractions/README.md

```diff
@@ -212,6 +212,34 @@ IChatClient client = new ChatClientBuilder()
 Console.WriteLine((await client.CompleteAsync("What is AI?")).Message);
 ```
 
+#### Options
+
+Every call to `CompleteAsync` or `CompleteStreamingAsync` may optionally supply a `ChatOptions` instance containing additional parameters for the operation. The most common parameters that are common amongst AI models and services show up as strongly-typed properties on the type, such as `ChatOptions.Temperature`. Other parameters may be supplied by name in a weakly-typed manner via the `ChatOptions.AdditionalProperties` dictionary.
+
+Options may also be baked into an `IChatClient` via the `UseChatOptions` extension method on `ChatClientBuilder`. This delegating client wraps another client and invokes the supplied delegate to populate a `ChatOptions` instance for every call. For example, to ensure that the `ChatOptions.ModelId` property defaults to a particular model name, code like the following may be used:
+```csharp
+using Microsoft.Extensions.AI;
+
+IChatClient client = new ChatClientBuilder()
+    .UseChatOptions(options =>
+    {
+        // If no model was specified...
+        if (options?.ModelId is null)
+        {
+            // Clone the options to avoid mutating the original instance, or create a new
```

Comment:
Unrelated to the docs per se, but this seems like a pitfall users can easily fall into. Would we consider creating a defensive copy before calling the delegate to avoid checks like this on the user's side?

Reply:
I've pushed back on it previously, but as the feedback has now been provided by multiple folks independently, I'll change it.